src/Stmt.h

```diff
@@ -559,9 +563,15 @@ class AssertStmt final : public Stmt
 
 private:
 	ExprPtr cond;
+	std::string cond_desc;
 	ExprPtr msg;
 	};
 
+// Helper function for reporting on asserts that either failed, or should
+// be processed regardless due to the presence of a "assertion_result" hook.
+extern void report_assert(bool cond, std::string_view cond_desc, StringValPtr msg_val,
```

Comment:
Thanks for the cleaning up that area.

It might be worth mentioning in the docstring that this raises an Exception for `cond == false` - this isn't very obvious at the call site anymore or by just looking at the header.

Reply:
Good thought, I've pushed that. (I mis-associated the fixup but not sure that merits a fixup-fixup.)